examples/ai_society/role_playing_with_role_generation.py

```diff
@@ -0,0 +1,109 @@
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =========== Copyright 2023 @ CAMEL-AI.org. All Rights Reserved. ===========
+from colorama import Fore
+
+from camel.agents.task_agent import RoleAssignmentAgent
+from camel.configs import ChatGPTConfig
+from camel.societies import RolePlaying
+from camel.typing import ModelType
+from camel.utils import print_text_animated
+
+AI_ASSISTANT_ROLE_INDEX = 0
+AI_USER_ROLE_INDEX = 1
+
+
+def main(model_type=None) -> None:
```

Comment:
Add a test to this eample.

Reply:
Thanks, I have added a `pytest` test in `examples\test\test_ai_society_with_role_generation.py`. But I am not sure if I do right. Your comments will be appreciated, thanks.

Reply:
@Appointat Check out the test for an exaple here: `examples\test\test_code_example.py`. It is as simple as calling main()

Reply:
Thanks, done : )